### ISSUE TITLE
Error: Error: sender doesn't have enough funds to send tx. The upfront cost is: 

### ISSUE THREAD

**Issue by vitaliy-kuzmich** [ConsenSys-archive/ganache-cli-archive#485]
Hey, it is not possible to specify amount of ether. It just does not works as described in docs. Thanks.

> #!/bin/bash

  # We define 10 accounts with balance 1M ether, needed for high-value tests.
   accounts=(
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501200,99999999999999999999999999999999999999999999999999999999999999"
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501201,99999999999999999999999999999999999999999999999999999999999999"
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501202,99999999999999999999999999999999999999999999999999999999999999"
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501203,99999999999999999999999999999999999999999999999999999999999999"
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501204,99999999999999999999999999999999999999999999999999999999999999"
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501205,99999999999999999999999999999999999999999999999999999999999999"
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501206,99999999999999999999999999999999999999999999999999999999999999"
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501207,99999999999999999999999999999999999999999999999999999999999999"
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501208,99999999999999999999999999999999999999999999999999999999999999"
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501209,99999999999999999999999999999999999999999999999999999999999999"
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501210,99999999999999999999999999999999999999999999999999999999999999"
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501211,99999999999999999999999999999999999999999999999999999999999999"
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501212,99999999999999999999999999999999999999999999999999999999999999"
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501213,99999999999999999999999999999999999999999999999999999999999999"
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501214,99999999999999999999999999999999999999999999999999999999999999"
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501215,99999999999999999999999999999999999999999999999999999999999999"
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501216,99999999999999999999999999999999999999999999999999999999999999"
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501217,99999999999999999999999999999999999999999999999999999999999999"
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501218,99999999999999999999999999999999999999999999999999999999999999"
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501219,99999999999999999999999999999999999999999999999999999999999999"
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501220,99999999999999999999999999999999999999999999999999999999999999"
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501221,99999999999999999999999999999999999999999999999999999999999999"
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501222,99999999999999999999999999999999999999999999999999999999999999"
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501223,99999999999999999999999999999999999999999999999999999999999999"
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501224,99999999999999999999999999999999999999999999999999999999999999"
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501225,99999999999999999999999999999999999999999999999999999999999999"
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501226,99999999999999999999999999999999999999999999999999999999999999"
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501227,99999999999999999999999999999999999999999999999999999999999999"
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501228,99999999999999999999999999999999999999999999999999999999999999"
    --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501229,99999999999999999999999999999999999999999999999999999999999999"
     --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501230,99999999999999999999999999999999999999999999999999999999999999"
     --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501231,99999999999999999999999999999999999999999999999999999999999999"
     --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501232,99999999999999999999999999999999999999999999999999999999999999"
     --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501233,99999999999999999999999999999999999999999999999999999999999999"
     --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501234,99999999999999999999999999999999999999999999999999999999999999"
     --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501235,99999999999999999999999999999999999999999999999999999999999999"
     --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501236,99999999999999999999999999999999999999999999999999999999999999"
     --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501237,99999999999999999999999999999999999999999999999999999999999999"
     --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501238,99999999999999999999999999999999999999999999999999999999999999"
     --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501239,99999999999999999999999999999999999999999999999999999999999999"
      --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501230,99999999999999999999999999999999999999999999999999999999999999"
          --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501240,99999999999999999999999999999999999999999999999999999999999999"
          --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501241,99999999999999999999999999999999999999999999999999999999999999"
          --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501242,99999999999999999999999999999999999999999999999999999999999999"
          --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501243,99999999999999999999999999999999999999999999999999999999999999"
          --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501244,99999999999999999999999999999999999999999999999999999999999999"
          --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501245,99999999999999999999999999999999999999999999999999999999999999"
          --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501246,99999999999999999999999999999999999999999999999999999999999999"
          --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501247,99999999999999999999999999999999999999999999999999999999999999"
          --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501248,99999999999999999999999999999999999999999999999999999999999999"
          --account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501249,99999999999999999999999999999999999999999999999999999999999999"

  )

/home/v/.npm-global/bin/testrpc "${accounts[@]}"

**Comment by vitaliy-kuzmich** [comment]
it displays correct amount before any transactions, but once at least 1 transaction executed, it sets the value to 100 ether...

**Comment by starlabs007** [comment]
This is pretty annoying, the `-e` (`--defaultBalanceEther`) doesn't work either. Looks like it defaults to 100 eth as well.

**Comment by mikeseese** [comment]
I think these things got fixed in https://github.com/trufflesuite/ganache-cli/pull/551

I'm going to close this due to inactivity, but if you still have issues with the latest versions (v6.1.6+), please open a new issue! Thanks!

**Comment by princesinha19** [comment]
I have same problem with the latest version. 
Error: sender doesn't have enough funds to send tx. The upfront cost is: 6000000000000000 and the sender's account only has: 0

**Comment by eccentricexit** [comment]
**Check your running processes and kill any odd truffle develop instances.**

I was having this issue on Xubuntu 18, truffle version:
```
Truffle v4.1.14 (core: 4.1.14)
Solidity v0.4.24 (solc-js)
```

To solve it, I:

1. Ran `top` or `htop` on a terminal and looked for `truffle develop` processes
2. Ran `kill [SIGKILL <PID>` to kill them.

Note that the standard kill signal (sigterm) may not be enough to kill the process if there are (for example) sockets open. This was my case, so in step 2 I used [sigkill](https://major.io/2010/03/18/sigterm-vs-sigkill/) instead. 

That solved it.
